docs/shopify_app/sessions.md

```diff
@@ -203,6 +204,66 @@ user.with_shopify_session do
 end
 ```
 
+#### Re-fetching an access token when API returns Unauthorized
+
+When using `ShopifyApp::EnsureHasSession` and the `new_embedded_auth_strategy` configuration, any **unhandled** Unauthorized `ShopifyAPI::Errors::HttpResponseError` will cause the app to perform token exchange to fetch a new access token from Shopify and the action to be retried once.
+
+```ruby
+class MyController < ApplicationController
+  include ShopifyApp::EnsureHasSession
+
+  def index
+    client = ShopifyAPI::Clients::Graphql::Admin.new(session: current_shopify_session)
+    client.query(options) # when this raises an Unauthorized error from Shopify, EnsureHasSession will retry the action once after performing token exchange
```

Comment:
Total nit, but having the comment after the line makes it a bit funky to read in MD mode (even at max width) because it's such a long line. Could we move them up / break them so it's easier to read without scrolling?

![image](https://github.com/Shopify/shopify_app/assets/64600052/a5ca4263-0c4c-417d-85e1-2470472396b1)